WalletWasabi.Fluent/ViewModels/Wallets/Send/SendViewModel.cs

```diff
@@ -240,10 +241,20 @@ private void ParseToField(string s)
 
 	private bool TryParseUrl(string? text)
 	{
+		if (_parsingTo)
+		{
+			return false;
+		}
+
+		_parsingTo = true;
```

Comment:
I don't keen on using a flag for this. We used to do it and had issues because in some cases it parsed the obsoleted string and just forget the most recent one because the parsing of the obsoleted was in progress.

Are you sure there is no other solution and it won't cause an issue again?

Reply:
@soosr

> it parsed the obsoleted string and just forget the most recent one 

 in this specific case, I think we actually *want* that to happen, because the parsing function is by itself changing the `To` value, but we do not want to discard the result obtained from the original parse, containing the PayJoin data:

1 - Paste PayJoin string
2 - `TryParseUrl()` is called, passing the PayJoin string as parameter
3 - `To = url.Address.ToString()` (line 282) sets `To` to a regular BTC address found inside the PayJoin string
4 - Change notification is raised for `To`
5 - Rx fires the `ParseToField()` passing the regular BTC address as parameter (this occurs in a deferred manner, so at this point the lock is already released)
6 - `TryParseUrl()` is called, passing the regular BTC address
7 - We must ignore the regular BTC address and keep the PayJoin data

Makes sense?